docs/tutorials/vqe.ipynb

```diff
@@ -350,11 +350,13 @@
     "\n",
     "\n",
     "def raw_callback(*args):\n",
-    "    job_id, (nfev, parameters, energy, stddev) = args\n",
-    "    intermediate_info[\"nfev\"].append(nfev)\n",
-    "    intermediate_info[\"parameters\"].append(parameters)\n",
-    "    intermediate_info[\"energy\"].append(energy)\n",
-    "    intermediate_info[\"stddev\"].append(stddev)"
+    "    # check if interim results, since both interim results (list) and final results (dict) are returned\n",
+    "    if type(args[1]) is list:\n",
```

Comment:
Very implicit check to tell interim and final results apart (hard to understand without the additional comment). Any chance the 
boolean `final` parameter passed to `user_messenger.publish` in the program can be reused here?

https://github.com/Qiskit/qiskit-ibm-runtime/blob/main/program_source/vqe/vqe.py#L1220

Reply:
Good idea, but I checked and final=True is not sent to the callback.